### PR TITLE
Visualizer - first commit

### PR DESCRIPTION
**Pull request by ulyssesp**
Eyyyoooo now you can visualize the mixtapes! There's only a couple
options right now, and it might need some design, but the code is all
there and vibing hard with the mixtape server.

Some notes:

- file-input-glvis.js is in vendor/assets/javascript. This is the main
javascript file. Construct the thing using FileInput and render it into
a dom element.

- shaders are in vendor/assets/shaders. The ones you want have to be added
manually to the FileInput constructor for now with a title and url.

- glvis.css is in vendor/assets/stylesheets. This is the only style
  sheet that should be included on the visualizer page.

### CODE REVIEW

**Comment by dpatti** [comment]
This is fantastic, I love it. I am seeing this error on the visualizer itself, though it seems to be working fine:

Failed to load resource: the server responded with a status of 404 (Not Found) http://localhost:3000/mixtapes/22/plane.vert

Also, it looks like you removed all the javascript from application.js accidentally. It needs to have mixtape.js and maybe even bootstrap.js.coffee included.

File: config/settings/development.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 contest:
   start: 2013-01-01T00:00:00.0
-  rotation: 2013-01-01T00:00:00.0
+  rotation: 2023-01-01T00:00:00.0
```
Comment (dpatti) [inline config/settings/development.yml]: Could you revert this?

**Comment by dpatti** [comment]
Did you need to put jquery in the vendor directory? It should be available automatically through the gem.

**Comment by ulyssesp** [comment]
[Seeking is an issue](http://stackoverflow.com/questions/27319054/how-to-stream-an-audio-file-without-a-direct-link) at least locally, but let's see how it is on the server. Are you running nginx?

**Comment by dpatti** [comment]
Seeking should be fine in production because nginx supports [range requests](https://tools.ietf.org/html/rfc7233) through `X-Sendfile`, which is what we use.

**Comment by ulyssesp** [comment]
Great!

File: app/views/mixtapes/edit.html.erb [hunk 2]
```diff
@@ -52,6 +52,7 @@
 
 <div id='current-upload'>
   <p>Drag one or more audio files here to add it to the mixtape. Drag them around with the handles to reorder them.</p>
+  <input type="file" id="input">
```
Comment (dpatti) [inline app/views/mixtapes/edit.html.erb]: What's this for?

**Comment by dpatti** [comment]
Other than that last question, this looks good

File: app/views/mixtapes/edit.html.erb (hunk 2 above)
Comment (ulyssesp) [inline app/views/mixtapes/edit.html.erb]: Shoot, left that in by accident. I was having a hard time dragging files w/o a file manager (linux).

**Comment by ulyssesp** [comment]
Cool. I'll squash and get it ready for merge.

**Comment by dpatti** [comment]
Could you squash those both into one commit for me?

**Comment by ulyssesp** [comment]
:+1:

**Comment by ulyssesp** [comment]
I have another shader I'd like to add. Separate PR? There'll probably be a couple new shaders and a couple updates to the js file over the next few week.s

**Comment by dpatti** [comment]
Yeah, feel free to PR as you add stuff. I've been thinking about removing the old listen entirely and just adding a blank shader too. I'll try and put this on prod soon so we can check it out.

**Comment by ulyssesp** [comment]
That's a good idea. We should double check it on a slower machine just in case. Even if the shader is blank, it'll still do all of the audio analysis.

**Comment by dpatti** [comment]
@ulyssesp Question: do you have your library in source control anywhere? I just want to know whether it would be unwise to make straight changes to the files you put in `/vendor`.

**Pull request merged**